packages/react/src/hooks/useAuth.ts

```diff
@@ -79,44 +61,56 @@ export const useAuth: UseAuth = () => {
   const { sessionId, userId, actor } = useAuthContext();
   const isomorphicClerk = useIsomorphicClerkContext();
 
-  const getToken: GetToken = createGetToken(isomorphicClerk);
-  const signOut: SignOut = createSignOut(isomorphicClerk);
+  const getToken: GetToken = useMemo(() => {
+    return createGetToken(isomorphicClerk);
+  }, [isomorphicClerk]);
+  const signOut: SignOut = useMemo(() => {
+    return createSignOut(isomorphicClerk);
+  }, [isomorphicClerk]);
+
+  // Make this stable as well!
```

Comment:
For this first iteration of this fix, I'd suggest focusing only on stabilizing `getToken`. This change might have undesired side-effects for useAuth use cases.

Reply:
That makes sense, specially considering probably most people destruct anyway so the value of stabilizing Auth itself is not so high; the reason I bundled both of the changes together is that I had a lot of trouble setting the whole testing and everything up, so since I was here I stabilized everything :)

> "This change might have undesired side-effects for useAuth use cases."

For a large project with so many tests I'd normally expect that if I make changes that I believe don't change the API, and no tests break, it should be fine. That's IMHO one of the main goals of tests, to be confident that changes+refactoring don't break things. (though I understand maybe these hooks are under-tested?)

Reply:
Fixed this as well! Now it's only memoizing (with `useCallback()`) the two functions, `signOut` and `getToken`, and not the whole return.